### ISSUE TITLE
Layout properties for icon behaves strange

### ISSUE THREAD

**Issue by AbrahamArun** [mapbox/mapbox-gl-js#2092]
```javascript
  let pinLayer = {
    'id':'pin',
    'type': 'symbol',
    'source':'location-pin',

    'layout': {
      'icon-image': 'map-pin'
      //'icon-allow-overlap': true,
      //'icon-ignore-placement': true
    },
    'interactive' : false,
    'minzoom': minZoom,
    'maxzoom': maxZoom
  };

map.addLayer(pinLayer);
```
On switching to the latest version 0.14.0, I have observed that the style breaks on ```icon-allow-overlap``` and ```icon-ignore-placement```.

On commenting those properties, the map looks like this (expected)
![screen shot 2016-02-09 at 5 53 02 pm](https://cloud.githubusercontent.com/assets/5391059/12916239/d5271842-cf56-11e5-932c-fe25cd5d2cfc.png)

And on adding those properties, the map renders like this
![screen shot 2016-02-09 at 5 53 34 pm](https://cloud.githubusercontent.com/assets/5391059/12916250/e7acd894-cf56-11e5-82b3-1022a3e229f1.png)

P.S.: I am using the style 'bright-v8'. And this is also the case with ```text-allow-overlap``` and ```text-ignore-placement```

**Comment by lucaswoj** [comment]
Any intuition on this @ansis?

**Comment by lucaswoj** [comment]
[Live demo](http://jsbin.com/weveluwemo/edit?html,output)

**Comment by AbrahamArun** [comment]
@lucaswoj That was neat, I was lazy to write a fiddle.

**Comment by lucaswoj** [comment]
This bug blames to b21b4c31c7a2407f558d4e73a830f399cb6ffb0d

**Comment by ansis** [comment]
I think we're not properly enabling and disabling the stencil test.

**Comment by lucaswoj** [comment]
@ansis Exactly. I already submitted a PR for your :eyes:

**Comment by lucaswoj** [comment]
the fix for this bug was just released in v0.14.1
